woocommerce/woocommerce
# Use syncpack to verify consistent versions for react and react-dom

### All Submissions:






### Changes proposed in this Pull Request:

This PR adds a GitHub action to run `syncpack` for `react` and `react-dom` dependencies. It also adds a pre-push hook to ensure that branches with version mismatches for either `react` or `react-dom` cannot be pushed.



Closes #34112.

### How to test the changes in this Pull Request:

1. Checkout this branch locally and then run `pnpm install`.
2. Create a new branch from this branch (`git checkout -b your-test-branch`).
3. Make a change to one of the updated `package.json` files updated in this PR to reverse one of the updated version numbers and commit it locally.
4. Attempt to push your branch to origin and verify that the pre-push hook prevents you from doing so.
5. Push your branch to origin with `--no-verify` and create a test PR with it against trunk.
6. Verify that the GitHub Action runs, fails, ~and creates a comment showing the files that mismatched~.
7. Run `pnpm exec syncpack -- fix-mismatches` locally. Commit and attempt pushing the changes.
8. Verify that the pre-push hook doesn't prevent you from pushing.
9. Verify that the action runs again, passes, ~and updates the comment~.

### Other information:




### FOR PR REVIEWER ONLY:
